core-services/personalOrganizer/src/main/java/org/silverpeas/core/personalorganizer/service/JournalDAO.java

```diff
@@ -130,40 +131,31 @@ public void removeJournal(Connection con, String id)
     }
   }
 
-  public boolean hasTentativeJournalsForUser(Connection con,
-      String userId) throws SQLException, java.text.ParseException {
-    PreparedStatement prepStmt = null;
-    ResultSet rs = null;
-    try {
-      prepStmt = getTentativePreparedStatement(con, userId);
-      rs = prepStmt.executeQuery();
-      return rs.next();
-    } finally {
-      DBUtil.close(rs, prepStmt);
+  public boolean hasTentativeJournalsForUser(Connection con, String userId) throws SQLException {
+    try (PreparedStatement prepStmt = getTentativePreparedStatement(con, userId)) {
+      try (ResultSet rs = prepStmt.executeQuery()) {
+        return rs.next();
+      }
     }
   }
 
   public Collection<JournalHeader> getTentativeJournalHeadersForUser(Connection con,
       String userId) throws SQLException, java.text.ParseException {
-    PreparedStatement prepStmt = null;
-    ResultSet rs = null;
-    List<JournalHeader> list = new ArrayList<JournalHeader>();
-    try {
-      prepStmt = getTentativePreparedStatement(con, userId);
-      rs = prepStmt.executeQuery();
-      while (rs.next()) {
-        JournalHeader journal = getJournalHeaderFromResultSet(rs);
-        list.add(journal);
+    List<JournalHeader> list = new ArrayList<>();
+    try (PreparedStatement prepStmt = getTentativePreparedStatement(con, userId)) {
+      try (ResultSet rs = prepStmt.executeQuery()) {
+        while (rs.next()) {
+          JournalHeader journal = getJournalHeaderFromResultSet(rs);
+          list.add(journal);
+        }
       }
-    } finally {
-      DBUtil.close(rs, prepStmt);
     }
     return list;
   }
 
   private PreparedStatement getTentativePreparedStatement(
       Connection con, String userId) throws SQLException {
-    String selectStatement = "select distinct " + JournalDAO.JOURNALCOLUMNNAMES
+    final String selectStatement = "select distinct " + JournalDAO.JOURNALCOLUMNNAMES
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "select distinct " 9 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)